<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h2>Noncompliant Code Example</h2>
<p>With a maximum number of 4 parameters:</p>
<pre>
func foo(p1 int, p2 int, p3 int, p4 int, p5 int) { // Noncompliant
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func foo(p1 int, p2 int, p3 int, p4 int) {
  // ...
}
</pre>

