<p>Most of the time a block of code is empty when a piece of code is really missing. So such empty block must be either filled or removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
func compute(a int, b int) {
	sum := a + b
	if  sum &gt; 0 { } // Noncompliant; empty on purpose or missing piece of code?
	fmt.Println("Result:", sum)
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func compute(a int, b int) {
	sum := a + b
	if  sum &gt; 0 {
		fmt.Println("Positive result")
	}
	fmt.Println("Result:", sum)
}
</pre>
<h2>Exceptions</h2>
<p>When a block contains a comment, this block is not considered to be empty. <code>for</code> and <code>select</code> statements with empty blocks
are ignored as well.</p>

