<p>There are several reasons for a method not to have a method body:</p>
<ul>
  <li> It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production. </li>
  <li> It is not yet, or never will be, supported. In this case an exception should be thrown. </li>
  <li> The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override. </li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
func doNothing() { // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func doNothing() {
  // Do nothing because of X and Y.
}
</pre>

