<p>Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.</p>
<p>On the other hand, constants can be referenced from many places, but only need to be updated in a single place.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold of 3:</p>
<pre>
func run() {
	prepare("foo")  // Noncompliant; 'foo' is duplicated 3 times
	execute("foo")
	release("foo")
}
</pre>
<h2>Compliant Solution</h2>
<pre>
const ACTION = "foo"

func run() {
	prepare(ACTION)
	execute(ACTION)
	release(ACTION)
}
</pre>
<h2>Exceptions</h2>
<p>To prevent generating some false-positives, literals having less than 5 characters are excluded.</p>

