<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste
error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical
operators, having the same value on both sides of an operator yields predictable results, and should be simplified.</p>
<h2>Noncompliant Code Example</h2>
<pre>
func main() {
  v1 := (true &amp;&amp; false) &amp;&amp; (true &amp;&amp; false) // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func main() {
  v1 := (true &amp;&amp; false) // Compliant
}
</pre>
<h2>Exceptions</h2>
<p>This rule ignores <code>*</code>, <code>+</code>, <code>&lt;&lt;</code> and <code>=</code>. </p>
<h2>See</h2>
<ul>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never
  executed </li>
  <li> {rule:go:S1656} - Implements a check on <code>=</code>. </li>
</ul>

